### PR TITLE
Include Spree::Metadata in Spree::User model and Setup CircleCI

### PR DESCRIPTION
**Pull request by aleksandarpetrushev**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,9 +1,159 @@
...
+  tests_postgres_ruby_3_0: &tests_postgres_ruby_3_0
+    <<: *run_tests_3_0
+    environment:
+      <<: *postgres_environment
+    docker:
+      - image: *ruby_3_0_image
+      - image: *postgres_image
+      - image: *redis_image
+
   update_staging_repository:
```
Comment (damianlegawiec) [inline .circleci/config.yml:151]: @aleksandarpetrushev this should be skipped

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,41 +1,162 @@
...
+    environment:
+      <<: *postgres_environment
+    docker:
+      - image: *ruby_3_0_image
+      - image: *postgres_image
+      - image: *redis_image
 
 workflows:
   version: 2
   update_staging_repository:
```
Comment (damianlegawiec) [inline .circleci/config.yml:153]:
```suggestion
  main:
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,41 +1,162 @@
...
+
+defaults_3_0: &defaults_3_0
+  <<: *defaults
+  docker:
+    - image: &ruby_3_0_image circleci/ruby:3.0-node-browsers
+    - image: *redis_image
+
+run_tests: &run_tests
+  <<: *defaults
+  parallelism: 8
```
Comment (damianlegawiec) [inline .circleci/config.yml:23]:
```suggestion
  parallelism: 2
```

more than 2 doesn't make sense for this repo

File: .circleci/config.yml [hunk 4]
```diff
@@ -1,41 +1,162 @@
...
+                            -o ~/rspec/rspec.xml \
+                            -- ${TESTFILES}
+    - store_test_results:
+        path: ~/rspec
+    - store_artifacts:
+        path: tmp/capybara
+
+run_tests_3_0: &run_tests_3_0
+  <<: *defaults_3_0
+  parallelism: 8
```
Comment (damianlegawiec) [inline .circleci/config.yml:56]:
```suggestion
  parallelism: 2
```

**Comment by aleksandarpetrushev** [comment]
@damianlegawiec  Should we add rspec_junit_formatter to gemfile?
bundler: failed to load command: rspec (/home/circleci/spree/vendor/bundle/ruby/2.7.0/bin/rspec)
LoadError: cannot load such file -- rspec_junit_formatter

**Comment by damianlegawiec** [comment]
> @damianlegawiec Should we add rspec_junit_formatter to gemfile? bundler: failed to load command: rspec (/home/circleci/spree/vendor/bundle/ruby/2.7.0/bin/rspec) LoadError: cannot load such file -- rspec_junit_formatter

yes :)

File: Gemfile [hunk 5]
```diff
@@ -5,6 +5,7 @@ gem 'spree', github: 'spree/spree', branch: 'main'
 gem 'spree_backend', github: 'spree/spree_backend', branch: 'main'
 gem 'spree_frontend', github: 'spree/spree_legacy_frontend', branch: 'main'
 gem 'spree_emails', github: 'spree/spree', branch: 'main'
+gem 'rspec_junit_formatter', '~> 0.4.1'
```
Comment (damianlegawiec) [inline Gemfile:8]: @aleksandarpetrushev let's add this please to `sptee_dev_tools` repo - https://github.com/spree/spree_dev_tools

**Comment by damianlegawiec** [comment]
@aleksandarpetrushev also let's re-enable mysql ;)

**Comment by damianlegawiec** [comment]
@aleksandarpetrushev also let's skip parallelism as this repo doesn't require it (too little tests)

File: Gemfile (hunk 5 above)
Comment (aleksandarpetrushev) [reply Gemfile:8]: Added https://github.com/spree/spree_dev_tools/commit/f4bb93916ec9e80379fae6e5aeffddf60dbb02cc

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**